### ISSUE TITLE
can ytt auto-explain why overlay matching did not match expected number of nodes

### ISSUE THREAD

**Issue by cppforlife** [k14s/ytt#503]
**Describe the problem/challenge you have**
Users may run into a challenge determining why their overlay does not match expected number of nodes. (e.g. if `overlay.subset({"apiVersion": "cert-manager.io/v1"})` but the actual node has `cert-manager.io/v1alpha1`). it would be interesting to somehow let the user know potential nodes that were "almost" matched.

no clue how to implement this.

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by pivotaljohn** [comment]
Perhaps a fuzzy match?  Like, "did you mean <closest potential match>?"

**Comment by cppforlife** [comment]
yeah, but how do we implement something like that. matchers could be arbitrarily complex in their logic (and we do not know what it is). does this only makes to try to implement for builtin matchers?
